VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -98,53 +98,35 @@ IOReturn VoodooI2CHIDDevice::parseHIDDescriptor() {
 }
 
 IOReturn VoodooI2CHIDDevice::getHIDDescriptorAddress() {
-    UInt32 guid_1 = 0x3CDFF6F7;
-    UInt32 guid_2 = 0x45554267;
-    UInt32 guid_3 = 0x0AB305AD;
-    UInt32 guid_4 = 0xDE38893D;
-    
-    OSObject *result = NULL;
-    OSObject *params[4];
-    char buffer[16];
-    
-    memcpy(buffer, &guid_1, 4);
-    memcpy(buffer + 4, &guid_2, 4);
-    memcpy(buffer + 8, &guid_3, 4);
-    memcpy(buffer + 12, &guid_4, 4);
-    
-    
-    params[0] = OSData::withBytes(buffer, 16);
-    params[1] = OSNumber::withNumber(0x1, 8);
-    params[2] = OSNumber::withNumber(0x1, 8);
-    params[3] = OSNumber::withNumber((unsigned long long)0x0, 8);
-    
-    acpi_device->evaluateObject("_DSM", &result, params, 4);
-    if (!result)
-        acpi_device->evaluateObject("XDSM", &result, params, 4);
-    if (!result) {
+    uuid_t guid;
+    uuid_parse(I2C_DSM_HIDG, guid);
+
+    // convert to mixed-endian
+    *(reinterpret_cast<uint32_t *>(guid)) = OSSwapInt32(*(reinterpret_cast<uint32_t *>(guid)));
+    *(reinterpret_cast<uint16_t *>(guid) + 2) = OSSwapInt16(*(reinterpret_cast<uint16_t *>(guid) + 2));
+    *(reinterpret_cast<uint16_t *>(guid) + 3) = OSSwapInt16(*(reinterpret_cast<uint16_t *>(guid) + 3));
+
+    UInt32 result;
+    OSObject *params[4] = {
+        OSData::withBytes(guid, 16),
+        OSNumber::withNumber(I2C_DSM_REVISION, 8),
+        OSNumber::withNumber(HIDG_DESC_INDEX, 8),
+        OSArray::withCapacity(1)
+    };
+
+    if (acpi_device->evaluateInteger("_DSM", &result, params, 4) != kIOReturnSuccess && acpi_device->evaluateInteger("XDSM", &result, params, 4) != kIOReturnSuccess) {
```

Comment:
Like the usage of `evaluateInteger`.
I still do prefer to split the `_DSM` and `XDSM` evaluation into separate conditions, as it's just too long to read..

Reply:
Maybe just a new line?